### DISCUSSION TITLE
wrk pressure test，8 cpu 16g ram，qps below 100k?

### DISCUSSION THREAD

**Discussion by laughingyear** [apache/apisix#5950] [Q&A]
I chang apisix/conf/config.yaml  ,add below to use port 7890:

nginx_config:
    main_configuration_snippet: |
        daemon on;
    http_configuration_snippet: |
        server
        {
            listen 7890;
            server_name _;
            access_log off;

            location / {
                default_type text/html;
                return 200 'This is text!';
            }
        }

        chunked_transfer_encoding on;

i add apisix route:

curl "http://127.0.0.1:9080/apisix/admin/upstreams/1" -H "X-API-KEY: edd1c9f034335f136f87ad84b625c8f1" -X PUT -d '
{
  "type": "roundrobin",
  "nodes": {
    "127.0.0.1:7890": 1
  }
}'

curl "http://127.0.0.1:9080/apisix/admin/routes/1" -H "X-API-KEY: edd1c9f034335f136f87ad84b625c8f1" -X PUT -d '
{
  "uri": "/",
  "host": "127.0.0.1",
  "upstream_id": "1"
}'

then i use wrk:
wrk -t8 -c10000 -d30 --latency http://127.0.0.1:9080/
Running 30s test @ http://127.0.0.1:9080/
  8 threads and 10000 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    39.25ms  161.09ms   1.97s    96.44%
    Req/Sec    10.24k     4.24k   25.63k    65.88%
  Latency Distribution
     50%    9.36ms
     75%   16.10ms
     90%   29.94ms
     99%    1.03s 
  2446920 requests in 30.10s, 410.63MB read
  Socket errors: connect 0, read 0, write 0, timeout 10656
  Non-2xx or 3xx responses: 157
Requests/sec:  81296.78
Transfer/sec:     13.64MB

command top shows:

![image](https://user-images.githubusercontent.com/86775317/147625969-8fa72f59-4076-49ea-9b44-6b5fde248384.png)

**Comment by tzssangglass** [comment]
Is your upstream, APISIX and wrk all on the same machine? Perhaps wrk and upstream are competing for system resources.

**Reply by laughingyear** [reply]
> Is your upstream, APISIX and wrk all on the same machine? Perhaps wrk and upstream are competing for system resources.

yeah，but cpu is only 60%

**Comment by tzssangglass** [comment]
What is your expected performance?

**Reply by laughingyear** [reply]
> What is your expected performance?
I think it should be 140K qps

**Reply by tzssangglass** [reply]
I would like to learn how this value is calculated.

BTW, here is the benchmark way: https://github.com/apache/apisix/blob/master/docs/en/latest/benchmark.md
